sdk/src/utils/abiCoder.test.ts

```diff
@@ -0,0 +1,101 @@
+import { encode, decode, decodeWithRetry } from "./abiCoder";
+import { encodeAbiParameters, decodeAbiParameters, parseAbiParameters } from "viem";
+
+jest.mock("viem", () => ({
+  encodeAbiParameters: jest.fn(),
+  decodeAbiParameters: jest.fn(),
+  parseAbiParameters: jest.fn(),
+}));
+
+describe("abiCoder", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe("encode", () => {
+    it("should encode the values with the given schema", () => {
+      const schema = "uint256";
+      const values = [123];
+      const mockEncodedData = "0xabc123";
+
+      (parseAbiParameters as jest.Mock).mockReturnValue([schema]);
+      (encodeAbiParameters as jest.Mock).mockReturnValue(mockEncodedData);
+
+      const result = encode(schema, values);
+
+      expect(parseAbiParameters).toHaveBeenCalledWith(schema);
+      expect(encodeAbiParameters).toHaveBeenCalledWith([schema], values);
+      expect(result).toBe(mockEncodedData);
+    });
+  });
+
+  describe("decode", () => {
+    it("should decode attestation data with the given schema", () => {
+      const schema = "uint256";
+      const attestationData = "0xabc123";
+      const mockDecodedData = [123];
+
+      (parseAbiParameters as jest.Mock).mockReturnValue([schema]);
+      (decodeAbiParameters as jest.Mock).mockReturnValue(mockDecodedData);
+
+      const result = decode(schema, attestationData);
+
+      expect(parseAbiParameters).toHaveBeenCalledWith(schema);
+      expect(decodeAbiParameters).toHaveBeenCalledWith([schema], attestationData);
+      expect(result).toBe(mockDecodedData);
+    });
+  });
+
+  describe("decodeWithRetry", () => {
```

Comment:
Can you add a few more cases to cover all possibilities?

- [ ] `reverseSchema` with this Schema: `score uint32`
- [ ] error in `decodeWrapped` (throwing from `parseAbiParameters` or `tryParse`, etc.)